Complete the editbar guide

mafanya23 opened the pull request:
Rewrites the Editbar guide (`docs/guides/diagram_editor/editbar.md`), which previously covered only the six `properties` groups, into a full guide on configuring Editbar controls. Also unifies the "Shapebar" spelling in the Shapebar guide.

## What's new in the guide

- **Initializing Editbar** and **Editbar controls** sections: the two ways of initialization, the `editbar` object properties, and the difference between basic and complex controls.
- **Configuring controls for Diagram elements**: the two ways of defining controls (by the type of a shape and by the group of elements), the priority of the shape type over the group, and the fact that a custom configuration replaces the default one of the same property.
- **Redefining properties of controls**: the `key` and `wrap` configs of basic controls, the `hidden`, `disabled` and `readOnly` configs, and the `$properties` service property of complex controls.
- **Configuring Editbar for the grid area / shapes / group elements / swimlanes / lines / line titles**: what controls suit each type of an element, with the related examples. The sections on groups and swimlanes explain how the controls address the nested `style`, `header`, `subHeaderCols` and `subHeaderRows` objects and which controls need redefining their keys.
- **Creating custom Editbar controls**: how a custom control inherits the configuration of the control specified in its `type` property, plus the descriptions of the `$on`, `$handler`, `$setValue` and `$layout` service properties.
- **Creating a dynamic Editbar**: a step-by-step instruction on modifying the controls dynamically, i.e. choosing the property to configure, specifying a function with the `item` and `editor` parameters, setting the conditions for rendering the controls, and the cases when the Editbar rebuilds its set of controls.
- **Using custom HTML in Editbar**, **Setting the width of Editbar**, **Showing/hiding the Editbar** and **Editbar appearance** sections.

## Corrections

- The example of the `$lineTitle` property used the `mindmap` mode, while line titles are available in the default mode only.
- The `$handler` service property is called on the events of the Form and its control, and not on the `change` event of DataCollection.
- The `$default` configuration is applied not only when there are no selected elements or several elements are selected, but also when the Diagram contains no data.
- A broken anchor link and the outdated links to the API pages.

mafanya23 committed:
[update] the editbar guide and properties config page

mafanya23 committed:
[update] the editbar guide

mafanya23 committed:
[update] the editbar and shapebar guides

mafanya23 committed:
[update] complete the editbar guide

Sorokin-Oleg commented on docs/guides/diagram_editor/editbar.md:
```diff
@@ -56,7 +184,8 @@ properties: {
 
-## Configure Editbar for shapes
+
+### Configuring Editbar for shapes
```
@mafanya23 хотелось бы дополнить этот раздел описанием, и сам пример по аналогии с примерами выше/ниже, то есть код целиком.

Sorokin-Oleg commented on docs/guides/diagram_editor/editbar.md:
```diff
@@ -6,39 +6,167 @@ description: You can learn about the Editbar of editor in the documentation of t
+
+**Related sample:** [Diagram Editor. Default mode. PERT сhart with the legend](https://snippet.dhtmlx.com/w8mrh3ay?mode=js)
+
+### Configuring Editbar for the grid area
```
@mafanya23 хотелось бы дополнить этот раздел описанием, и сам пример по аналогии с примерами выше/ниже, то есть код целиком.

Sorokin-Oleg commented on docs/guides/diagram_editor/editbar.md:
```diff
@@ -77,7 +206,7 @@ properties: {
 **Related complex controls:** [Border](/api/diagram_editor/editbar/complex_controls/border/), [Arrange](/api/diagram_editor/editbar/complex_controls/arrange/), [Position](/api/diagram_editor/editbar/complex_controls/position/), [Size](/api/diagram_editor/editbar/complex_controls/size/), [Text align](/api/diagram_editor/editbar/complex_controls/textalign/), [Text style](/api/diagram_editor/editbar ...
 
-## Configure Editbar for group elements
+### Configuring Editbar for group elements
```
@mafanya23 хотелось бы дополнить этот раздел описанием, и сам пример по аналогии с примерами выше/ниже, то есть код целиком.

Sorokin-Oleg commented on docs/guides/diagram_editor/editbar.md:
```diff
@@ -95,7 +224,7 @@ properties: {
 **Related complex controls:** [Border](/api/diagram_editor/editbar/complex_controls/border/), [Arrange](/api/diagram_editor/editbar/complex_controls/arrange/), [Header](/api/diagram_editor/editbar/complex_controls/header/), [Header common](/api/diagram_editor/editbar/complex_controls/headercommon/), [Header position](/api/diagram_editor/editbar/complex_controls/headerposition/), [Size](/api/diagr ...
 
-## Configure Editbar for swimlanes
+### Configuring Editbar for swimlanes
```
@mafanya23 хотелось бы дополнить этот раздел описанием, и сам пример по аналогии с примерами выше/ниже, то есть код целиком.

Sorokin-Oleg commented on docs/guides/diagram_editor/editbar.md:
```diff
@@ -113,7 +242,7 @@ properties: {
 **Related complex controls:** [Border](/api/diagram_editor/editbar/complex_controls/border/), [Arrange](/api/diagram_editor/editbar/complex_controls/arrange/), [Header](/api/diagram_editor/editbar/complex_controls/header/), [Header common](/api/diagram_editor/editbar/complex_controls/headercommon/), [Header position](/api/diagram_editor/editbar/complex_controls/headerposition/), [Size](/api/diagr ...
 
-## Configure Editbar for lines
+### Configuring Editbar for lines
```
@mafanya23 хотелось бы дополнить этот раздел описанием, и сам пример по аналогии с примерами выше/ниже, то есть код целиком.

Sorokin-Oleg commented on docs/guides/diagram_editor/editbar.md:
```diff
@@ -131,13 +260,13 @@ properties: {
 **Related complex controls:** [Border](/api/diagram_editor/editbar/complex_controls/border/), [Line shape](/api/diagram_editor/editbar/complex_controls/lineshape/), [Pointer view](/api/diagram_editor/editbar/complex_controls/pointerview/)
 
-## Configure Editbar for line titles
+### Configuring Editbar for line titles
```
@mafanya23 хотелось бы дополнить этот раздел описанием, и сам пример по аналогии с примерами выше/ниже, то есть код целиком.

Sorokin-Oleg commented on docs/guides/diagram_editor/editbar.md:
```diff
@@ -150,14 +279,240 @@ properties: {
+There is a possibility to create an Editbar that will dynamically change depending on certain conditions, e.g.:
+the selected shape type, absence of selected items, the properties of the selected item. Check the example:
+
+~~~jsx
```
@mafanya23 этот пример не про динамическое отображение Editbar, скорей про базовую конфигурацию и переопределение определенных типов элементов (фигур)

mafanya23 replied:
Убрала

Sorokin-Oleg commented on docs/guides/diagram_editor/editbar.md:
```diff
@@ -150,14 +279,240 @@ properties: {
+To make a dynamic Editbar that will change *depending on the properties of a selected shape*, you need to specify a function as a value of the property that defines the [type of a shape](/shapes/default_shapes/) or a [group of items](/guides/items_index/) within the [`properties`](api/diagram_editor/editbar/config/properties_property.md) configuration object:
+
+~~~jsx 
+editbar: {
```
@mafanya23 тут можно привести весь конфиг, он не большой. При этом было бы не плохо описать, что происходит в примере. Если очень кратко, то метод `hasOwnProperty` проверяет на наличие в объекте собственного свойства, указанного в качестве аргумента. И когда условие выполняется, мы отображаем определенный контрол.  Например, в объекте есть свойство `title`, тогда мы отобразим контрол, который будет отвечать за отображение/изменение данного свойства. С остальными по аналогии, будет так же.

Sorokin-Oleg replied:
Заметил описание есть пониже, возможно стоит чуть больше разбавить?)

mafanya23 replied:
Обновила

Sorokin-Oleg commented on docs/guides/diagram_editor/editbar.md:
```diff
@@ -150,14 +279,240 @@ properties: {
+
+In the above example the `$shape` property is specified as a function that provides a set of default controls for all Diagram shapes and allows using additional controls, depending on whether a Diagram item has a particular property in its configuration.
+
+## Setting the width of Editbar
```
@mafanya23 в задаче есть еще два не плохих примера которые можно показать с описанием. Хотелось бы рассказать о них в гайде. Покажут как пользоваться сервисными свойствами.
![image](https://github.com/user-attachments/assets/ff4bc2a5-a4ad-4329-bbea-8cc86a676e27)

Sorokin-Oleg replied:
Там где про сервисные свойства, нужно таблички с предупреждениями показать дополнительно

mafanya23 replied:
Добавила раздел Redefining service properties of basic controls в главу про создание кастомных контролов

mafanya23 committed:
Merge branch 'next' into mr-next-editbar-guide-4591

mafanya23 commented on docs/guides/diagram_editor/editbar.md:
```diff
@@ -6,39 +6,167 @@ description: You can learn about the Editbar of editor in the documentation of t
-- [`$line`](#configure-editbar-for-lines) allows configuring Editbar controls for all elements with the [**line**](/lines/) type
-- [`$lineTitle`](#configure-editbar-for-line-titles) - allows configuring Editbar controls for all elements with the [**lineTitle**](/line_titles/) type
+- The `colorpicker`, `spacer` and `button` controls are used for all the Diagram items with the `shape` type
+    - The configs of the `colorpicker` and `button` controls are redefined especially for this type of Diagram items
```
Да, наверное, здесь лучше написать especially for the $shape group of Diagram items, i.e. for all shapes

mafanya23 committed:
[update] corrections in the editbar guide

mafanya23 committed:
[update] complete the editbar guide

mafanya23 committed:
[update] minor corrections in the editbar guide

mafanya23 committed:
[update] merge next into editbar guide branch

- resolved conflicts in the editbar and shapebar guides
- editbar: kept the rewritten guide, applied the link normalization
  and formatting fixes from next across the whole file
- shapebar: took the updated links and grammar from next, kept the
  "Shapebar" capitalization from the branch

mafanya23 committed:
[update] extend the dynamic Editbar guide

- describe the parameters and the return value of a function set
  as a value of the properties config
- note that such a function can be set for any properties key,
  including the type of a shape and $default
- explain that the configuration of a shape type replaces the one
  of its group and that a custom config replaces the default one
- specify when the Editbar rebuilds its set of controls and that
  modifying a selected item only refreshes the control values
- mention that the built-in $shape config is a function as well
- fix the broken anchor link, the events that call $handler and
  the list of cases when $default is applied

mafanya23 committed:
[update] restructure the dynamic Editbar guide

- split the section into 4 steps: choosing a property, specifying
  a function, setting conditions, checking the Editbar rebuilding
- move the general rules before the $shape example
- specify which methods allow checking the state of the editor
  within a function set for the $default property
- extend the $default example with the selection and data checks
- link the API pages of the Diagram Editor and of the used methods

mafanya23 committed:
[update] detail the configuring of Editbar controls

- describe what controls suit the grid area, shapes, groups,
  swimlanes, lines and line titles, with the related examples
- specify how the controls address the nested properties of groups
  and swimlanes and which of them need redefining their keys
- explain the key and wrap configs of basic controls and merge the
  sections on redefining basic and complex controls
- note that a button works only with the $on or $handler property
- fix the mode of the editor in the line title example, since line
  titles are available in the default mode only

mafanya23 committed:
[update] revise the Editbar guide

- correct the description of the properties config, which sets
  the controls per type of an element and not per control property
- specify the hidden, disabled and readOnly configs of controls
- note that a custom control inherits the config of the control
  specified in its type property, including the key config
- rename the section on service properties and correct the wording
  of the sentences on the grid area and the custom HTML content
